### PR TITLE
Fix for ExprValueFactory construct issue

### PR DESCRIPTION
**Pull request by rupal-bq**
*Issue #858, #867*

*Description of changes:*
- added date, time, and datetime in ExprValueFactory
- added unit tests 
- added manual integration tests (SQLite returned the wrong type in a schema and the comparison test failed).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dai-chen** [comment]
Is https://github.com/opendistro-for-elasticsearch/sql/issues/867 same issue?

**Comment by rupal-bq** [comment]
> Is #867 same issue?

Thanks for noticing, it's the same issue. I will add a test for that case as well.

**Comment by chloe-zh** [comment]
Could you also add test cases in comparison test?

**Review by chloe-zh — APPROVED** [review]

File: elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/data/value/ElasticsearchExprValueFactory.java [hunk 1]
```diff
@@ -177,14 +183,22 @@ public ExprValue construct(String field, Object value) {
       return constructString((String) value);
     } else if (type.equals(BOOLEAN)) {
       return constructBoolean((Boolean) value);
-    } else if (type.equals(TIMESTAMP)) {
+    } else if (type.equals(TIMESTAMP) || type.equals(DATE) || type.equals(TIME)
```
Comment (penghuo) [inline elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/data/value/ElasticsearchExprValueFactory.java:186]: Quesiton,  In which case the field will in type of DATA/TIME?
The main purpose of ExprValueFactory is extract data from Elasticsearch response. Elasticsearch only support date type and it is mapped to TIMESTAMP.

Comment (penghuo) [reply elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/data/value/ElasticsearchExprValueFactory.java:186]: Never mind, this is the aggregation result. I see the problem.

**Review by penghuo — APPROVED** [review]
Thanks for the change.

**Pull request merged**